tsconfig.base.json

```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
+    "lib": [
+      "ESNext"
+    ],
+    "module": "esnext",
+    "target": "esnext",
+    "moduleResolution": "nodenext",
+    "strict": true,
```

Comment:
I think `strict` for runtime tests will end up being a little bit of a velocity hit for us

Reply:
The changes required to modify existing tests compatible with strict mode were pretty minimal: https://github.com/oven-sh/bun/pull/1475/commits/737ef3c9f7879599f5faf9441238ea8a4b5c3d9f

Also part of the goal here is for the test suite to sanity check the typings as well, so this is necessary if we want the tests to reflect a realistic user environment.